pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go

```diff
@@ -213,11 +241,143 @@ func createLogicalReplicationStreamTypeCheck(
 	}
 
 	return true, streamCreationHeader, nil
 }
+
+type resolvedLogicalReplicationOptions struct {
+	cursor          *hlc.Timestamp
+	mode            *string
+	defaultFunction *jobspb.LogicalReplicationDetails_DefaultConflictResolution
+	// Mapping of table name to function descriptor
+	userFunctions map[string]int32
+}
+
+func evalLogicalReplicationOptions(
+	ctx context.Context,
+	options tree.LogicalReplicationOptions,
+	eval exprutil.Evaluator,
+	p sql.PlanHookState,
+) (*resolvedLogicalReplicationOptions, error) {
+	r := &resolvedLogicalReplicationOptions{}
+	if options.Mode != nil {
+		mode, err := eval.String(ctx, options.Mode)
+		if err != nil {
+			return nil, err
+		}
+		r.mode = &mode
+	}
+	if options.Cursor != nil {
+		cursor, err := eval.String(ctx, options.Cursor)
+		if err != nil {
+			return nil, err
+		}
+		asOfClause := tree.AsOfClause{Expr: tree.NewStrVal(cursor)}
+		asOf, err := asof.Eval(ctx, asOfClause, p.SemaCtx(), &p.ExtendedEvalContext().Context)
+		if err != nil {
+			return nil, err
+		}
+		r.cursor = &asOf.Timestamp
+	}
+	if options.DefaultFunction != nil {
+		defaultResolution := &jobspb.LogicalReplicationDetails_DefaultConflictResolution{}
+		defaultFnc, err := eval.String(ctx, options.DefaultFunction)
+		if err != nil {
+			return nil, err
+		}
+		switch strings.ToLower(defaultFnc) {
+		case "lww":
+			defaultResolution.ConflictResolutionType = jobspb.LogicalReplicationDetails_DefaultConflictResolution_LWW
+		case "dlq":
+			defaultResolution.ConflictResolutionType = jobspb.LogicalReplicationDetails_DefaultConflictResolution_DLQ
+		// This case will assume that a function name was passed in
+		// and we will try to resolve it.
+		default:
+			urn := tree.MakeUnresolvedName(defaultFnc)
+			descID, err := lookupFunctionID(ctx, p, urn)
+			if err != nil {
+				return nil, err
+			}
+			defaultResolution.ConflictResolutionType = jobspb.LogicalReplicationDetails_DefaultConflictResolution_UDF
+			defaultResolution.FunctionId = descID
+		}
+
+		r.defaultFunction = defaultResolution
+	}
+	if options.UserFunctions != nil {
+		r.userFunctions = make(map[string]int32)
+		for tb, fnc := range options.UserFunctions {
+			objName, err := tb.ToUnresolvedObjectName(tree.NoAnnotation)
+			if err != nil {
+				return nil, err
+			}
+
+			urn := tree.MakeUnresolvedName(fnc.String())
+			descID, err := lookupFunctionID(ctx, p, urn)
+			if err != nil {
+				return nil, err
+			}
+			r.userFunctions[objName.String()] = descID
+		}
+	}
+	return r, nil
+}
+
+func lookupFunctionID(
+	ctx context.Context, p sql.PlanHookState, u tree.UnresolvedName,
+) (int32, error) {
+	rf, err := p.ResolveFunction(ctx, tree.MakeUnresolvedFunctionName(&u), &p.SessionData().SearchPath)
+	if err != nil {
+		return 0, err
+	}
+	if len(rf.Overloads) > 1 {
+		return 0, errors.Newf("function '%s' has more than 1 overload", u.String())
+	}
+	fnOID := rf.Overloads[0].Oid
+	descID := typedesc.UserDefinedTypeOIDToID(fnOID)
+	return int32(descID), nil
+}
```

Comment:
@stevendanna So this lookup works based on our slack conversation. 
```
demo@127.0.0.1:26257/demoapp/movr> CREATE LOGICAL REPLICATION STREAM FROM TABLE users ON                                                                                                 
                                -> 'postgresql://demo:demo13904@127.0.0.1:26258/movr?options=-ccluster%3Ddemoapp&sslmode=require&sslrootcert=%2FUsers%2Fjeremyyang%2F.cockroach-demo%2Fca
                                -> .crt' INTO TABLE users WITH FUNCTION add FOR TABLE users;                                                                                             
[]jobspb.LogicalReplicationDetails_ReplicationPair{jobspb.LogicalReplicationDetails_ReplicationPair{SrcDescriptorID:106, DstDescriptorID:106, SrcFunctionID:112}}
```

Its now plumbed into the existing replication pairs details that already exists. Does that look useable for the UDF work you have?